File: src/main/java/net/hectus/neobb/turn/default_game/block/TStonecutter.java
package net.hectus.neobb.turn.default_game.block;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.CounterFilter;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.function.CounterFunction;
import net.hectus.neobb.util.Modifiers;
import org.bukkit.block.Block;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class TStonecutter extends BlockTurn implements CounterFunction, NeutralClazz {
    public TStonecutter(NeoPlayer player) { super(player); }
    public TStonecutter(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 4;
    }

    @Override
    public List<CounterFilter> counters() {
        return List.of(CounterFilter.of(turn -> turn.getClass().getSimpleName().endsWith("Wall"), "walls"));
    }

    @Override
    public void counter(@NotNull NeoPlayer source) {
        source.opponents(true).forEach(p -> p.removeModifier(Modifiers.P_DEFAULT_DEFENDED));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/block/TStonecutter.kt
package net.hectus.neobb.modes.turn.default_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.Turn
import net.hectus.neobb.modes.turn.default_game.CounterFilter
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.CounterFunction
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.util.Modifiers
import org.bukkit.block.Block

class TStonecutter(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), CounterFunction, NeutralClazz {
    override val cost: Int = 4

    override fun counters(): List<CounterFilter> {
        return listOf(CounterFilter.of("walls") { it::class.simpleName!!.endsWith("Wall") })
    }

    override fun counter(source: NeoPlayer, countered: Turn<*>) {
        super.counter(source, countered)
        source.opponents().forEach { it.removeModifier(Modifiers.Player.Default.DEFENDED) }
    }
}
